File: Data Structures & Algorithms/task-scheduling/submission-0.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
     unordered_map<char,int>mpp;
     for(char it:tasks){
        mpp[it]++;
     }   
     priority_queue<int>maxHeap;
     queue<pair<int,int>>q;

     for(auto it:mpp){
      maxHeap.push(it.second);


     }

     int time=0;
     
     while(!maxHeap.empty()||!q.empty()){

      if(!q.empty()&&time>=q.front().second){
        
        maxHeap.push(q.front().first);

        q.pop();





      }
      
      if(!maxHeap.empty()){
        int cnt=maxHeap.top()-1;


        maxHeap.pop();

        if(cnt>0){
          q.push({cnt,time+1+n});

        }  
 



      }
      time++;






     }

     return time;




    }
};

File: Data Structures & Algorithms/average-waiting-time/submission-0.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        
        double time=0;
        int cur=customers[0][0];
        for(int i=0;i<customers.length;i++){
            
            int ar=customers[i][0];
            
            int ti=customers[i][1];
            
            if(cur<ar){
               
               cur=ar;

            }
              cur+=ti;

            time+=(double)(cur-ar);
            System.out.println(cur+" "+time);



        }

        return (double)time/customers.length;





    }
}